lib/spack/spack/fetch_strategy.py

```diff
@@ -57,6 +57,15 @@
 from spack.util.compression import decompressor_for, extension
 
 import spack.util.pattern as pattern
+
+# The xmlrpclib module has been renamed to xmlrpc.client in Python 3
+# Spack does not yet support Python 3, but there's no reason not to be
+# future-proof
```

Comment:
Would it be better to use `six` here, rather than rolling our own Python2-vs-Python3 compatibility?  `six` is all in one file and MIT licensed, really easy (and legal) to just include with Spack.